### PR TITLE
Simplify mask implementation

### PR DESCRIPTION
**Pull request by vwxyzjn**
This PR makes simplify the masking implementation per findings in https://github.com/kachayev/gym-microrts-paper-sb3.

@kachayev made this helpful experiment showing masking the logits is enough. This PR does this (no longer overwriting the entropy function in `CategoricalMasked`) and caches the `mask_value = torch.tensor(-1e8)`

image

### CODE REVIEW

**Comment by vwxyzjn** [comment]
Merging this now. @kered9 @xluox @kachayev.
